src/pages/PageGenre.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div>
+    <tabs-music></tabs-music>
+
+    <content-with-heading>
+      <template slot="heading-left">
+        <p class="title is-4">{{ name }}</p>
+        <p class="heading">{{ genreAlbums.total }} albums</p>
+      </template>
+      <template slot="heading-right">
+        <a class="button is-small is-dark is-rounded" @click="play">
+          <span class="icon">
+            <i class="mdi mdi-play"></i>
+          </span>
+          <span>Play</span>
+        </a>
+      </template>
+      <template slot="content">
+        <list-item-albums v-for="album in genreAlbums.items" :key="album.id" :album="album"></list-item-albums>
+      </template>
+    </content-with-heading>
+  </div>
+</template>
+
+<script>
+import { LoadDataBeforeEnterMixin } from './mixin'
+import ContentWithHeading from '@/templates/ContentWithHeading'
+import TabsMusic from '@/components/TabsMusic'
+import ListItemAlbums from '@/components/ListItemAlbum'
+import webapi from '@/webapi'
+
+const PageGenre = {
+  data () {
+    return {
+      name: ''
+    }
+  },
+
+  load: function (to) {
+    this.name = to.params.genre
+    return webapi.library_genre(to.params.genre)
+  },
+
+  set: function (vm, response) {
+    vm.name = this.name
+    vm.genreAlbums = response.data.albums
+  }
+}
+
+export default {
+  name: 'PageGenrex',
+  mixins: [ LoadDataBeforeEnterMixin(PageGenre) ],
+  components: { ContentWithHeading, TabsMusic, ListItemAlbums },
+
+  data () {
+    return {
+      name: '',
+      genreAlbums: {}
+    }
+  },
+
+  methods: {
+    queueem: function () {
+      var i
+      for (i = 0; i < this.genreAlbums.items.length; i++) {
+        webapi.queue_add(this.genreAlbums.items[i].uri)
```

Comment:
This only works, because forked-daapd uses a single thread to process json api requests. The calls to the webapi are asynchronous and it is not guaranteed in which order they are executed. 

That is why the calls to the webapi need to be chained, so that the next one is executed after the current one returned.

To reduce the number of requests to the server, it is better to concatenate the album uris and make a single call to queue/add.

```
play: function () {
      webapi.queue_clear().then(() =>
        webapi.queue_add(this.genreAlbums.items.map(a => a.uri).join(',')).then(() =>
          webapi.player_play()
        )
      )
    }
```